src/apps/search-header/search-header.tsx

```diff
@@ -1,27 +1,160 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
+import { useCombobox, UseComboboxStateChange } from "downshift";
+import {
+  SuggestionsFromQueryStringQuery,
+  useSuggestionsFromQueryStringQuery
+} from "../../core/dbc-gateway/generated/graphql";
 import SearchBar from "../../components/search-bar/search-bar";
+import { Autosuggest } from "../../components/autosuggest/autosuggest";
+import { Suggestion } from "../../components/autosuggest-text/autosuggest-text-item";
 
 export interface SearchHeaderProps {
   searchHeaderUrl?: string;
   altText?: string;
-  inputPlaceholder?: string;
+  inputPlaceholderText?: string;
+  stringSuggestionAuthorText?: string;
+  stringSuggestionWorkText?: string;
+  stringSuggestionTopicText?: string;
 }
 
 const SearchHeader: React.FC<SearchHeaderProps> = ({
   searchHeaderUrl = "/search",
   altText = "search icon",
-  inputPlaceholder = "Search here"
+  inputPlaceholderText = "Search here",
+  stringSuggestionAuthorText = "Author",
+  stringSuggestionWorkText = "Work",
+  stringSuggestionTopicText = "Topic"
 }) => {
-  const [q, setQ] = useState("");
+  const [q, setQ] = useState<string>("");
+  const [suggestItems, setSuggestItems] = useState<any[]>([]);
+  const [currentlySelectedItem, setCurrentlySelectedItem] = useState<any>("");
+  const [isAutosuggestOpen, setIsAutosuggestOpen] = useState<boolean>(false);
+  const {
+    data,
+    isLoading,
+    status
+  }: {
+    data: SuggestionsFromQueryStringQuery | undefined;
+    isLoading: boolean;
+    status: string;
+  } = useSuggestionsFromQueryStringQuery({ q });
+
+  // once the query returns data, we set it into our useSate
+  useEffect(() => {
+    if (data) {
+      const arayOfResults = data.suggest.result;
+      setSuggestItems(arayOfResults);
+    }
+  }, [data]);
+
+  // if there are at least 3 chars in the search-field we open the autosuggest
+  // by design this is how the autosuggest should work
+  useEffect(() => {
+    if (q) {
+      if (q.length > 2) {
+        setIsAutosuggestOpen(true);
+      }
+    } else {
+      setIsAutosuggestOpen(false);
+    }
+  }, [q]);
+
+  function determinSuggestionType(suggestion: Suggestion): string {
+    switch (suggestion.__typename) {
+      case "Creator":
+        return suggestion.name;
+      case "Subject":
+        return suggestion.value;
+      default:
+        return suggestion.title !== null && suggestion.title !== undefined
+          ? suggestion.title
+          : "incomplete data";
```

Comment:
Not sure I am reading this correct,  but will "incomplete data" ever be presented to the user?

Reply:
Yes it will. In the case of the API returning a title that is either null or undefined, the handleSelectedItemChange function calls for determinSuggestionType which will return this string ("incomplete data"). As long as the title is not null or undefined, it will just be used as it comes from the API :)